scipy/optimize/tests/test__simulated_annealing.py

```diff
@@ -0,0 +1,115 @@
+# Simulated dual annealing unit tests implementation.
```

Comment:
The test suite does not have comprehensive coverage , with various tests missing. E.g.

- testing that maxfun works
- testing that maxiter works
- testing that the args tuple is passed correctly
- testing correct and incorrect bounds
- `VisitingDistribution` and `MarkovChain`, `EnergyState` are missing comprehensive testing.
- etc

As an example, I'd hope that something approximating the depth of `optimize.tests.test__differential_evolution` could be approached.